salesforce/metadataApi.go

```diff
@@ -0,0 +1,268 @@
+package salesforce
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"log/slog"
+	"net/http"
+	"net/url"
+	"strings"
+
+	"github.com/amp-labs/connectors/common"
+	"golang.org/x/oauth2"
+)
+
+const (
+	versionPrefix             = "v"
+	openParenthesis           = "<"
+	closeParenthesis          = ">"
+	closeWithSlashParenthesis = "/>"
+)
+
+var ErrCreateMetadata = fmt.Errorf("error in CreateMetadata")
+
+type XMLSchema interface {
```

Comment:
We should extract the XML helpers to another file or lib & only have the main connector method logic in this file.